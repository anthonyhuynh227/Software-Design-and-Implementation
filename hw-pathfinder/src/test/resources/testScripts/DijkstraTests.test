#find shortestPath with Dijkstra algorithm and some edge cases.

#create graphs
CreateGraph A
CreateGraph B
CreateGraph C

#Add nodes into graphs
AddNode A a1
AddNode B b1
AddNode B b2
AddNode C c1
AddNode C c2
AddNode C c3

#Add edges to graphs
AddEdge B b1 b2 1.000
AddEdge B b2 b1 3.000
AddEdge C c1 c2 5.000
AddEdge C c2 c3 7.000
AddEdge C c1 c3 10.000

#Print all nodes in the graph
ListNodes A
ListNodes B
ListNodes C

#Print all the child nodes with outgoing edge
ListChildren A a1
ListChildren B b1
ListChildren B b2
ListChildren C c1
ListChildren C c2
ListChildren C c3

#Test one node is not in graph
FindPath A a1 a2

#Test two nodes are not in graph
FindPath A a2 a3

#Test path exist in graph
FindPath B b1 b2

FindPath B b2 b1

#Test path from a node to itself
FindPath B b1 b1

#Test no path in graph
FindPath C c3 c1

#Test shortest path
FindPath C c1 c3