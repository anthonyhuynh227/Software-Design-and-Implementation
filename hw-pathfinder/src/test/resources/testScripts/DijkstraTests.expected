#find shortestPath with Dijkstra algorithm and some edge cases.

#create graphs
created graph A
created graph B
created graph C

#Add nodes into graphs
added node a1 to A
added node b1 to B
added node b2 to B
added node c1 to C
added node c2 to C
added node c3 to C

#Add edges to graphs
added edge 1.000 from b1 to b2 in B
added edge 3.000 from b2 to b1 in B
added edge 5.000 from c1 to c2 in C
added edge 7.000 from c2 to c3 in C
added edge 10.000 from c1 to c3 in C

#Print all nodes in the graph
A contains: a1
B contains: b1 b2
C contains: c1 c2 c3

#Print all the child nodes with outgoing edge
the children of a1 in A are:
the children of b1 in B are: b2(1.000)
the children of b2 in B are: b1(3.000)
the children of c1 in C are: c2(5.000) c3(10.000)
the children of c2 in C are: c3(7.000)
the children of c3 in C are:

#Test one node is not in graph
unknown: a2

#Test two nodes are not in graph
unknown: a2
unknown: a3

#Test path exist in graph
path from b1 to b2:
b1 to b2 with weight 1.000
total cost: 1.000

path from b2 to b1:
b2 to b1 with weight 3.000
total cost: 3.000

#Test path from a node to itself
path from b1 to b1:

#Test no path in graph
path from c3 to c1:
no path found

#Test shortest path
path from c1 to c3:
c1 to c3 with weight 10.000
total cost: 10.000